src/code/engine_library/plotReportTimeProfile.m

```diff
@@ -91,78 +108,87 @@
 
 
-% legend handle array for plots
-lgh=[];
+% Patch of reference poulation
+% Plot before main population, so it is in background
+if ~isnan(yMinRef)
+    
+    % exclude log(0)
+     ij = find(~isinf(yMinRef) & ~isinf(yMaxRef)); 
+
+    lgh(4) = patch([SimTL.timeRef(ij), SimTL.timeRef(ij(end:-1:1))],[yMinRef(ij), yMaxRef(ij(end:-1:1))],WSettings.colorVector(2,:),...
+        'linestyle','-','LineWidth',0.5,'FaceAlpha',WSettings.FaceAlpha,...
+        'EdgeColor',WSettings.colorVector(2,:),'displayname',strtrim(strrep(legendEntries{2},'<xxx>',rangeTxt)));
     
+    % Mean of referencep population 
+    lgh(3)=plot(SimTL.timeRef,yMeanRef,'-','color',WSettings.colorVector(2,:),'linewidth',2,...
+        'displayname',strtrim(strrep(legendEntries{2},'<xxx>',legendTextMean)));
+
+end
+
+
+
 % if not single plot patch
 if ~isnan(yMin)   
     
-    % exclude log(0)
-    ij = find(~isinf(yMin) & ~isinf(yMax)); 
+    % exclude values below ylim
+    jj = yMin < ylim;
+    yMin(jj) = ylim;
+    ij = find(yMax>=ylim); 
     
     
-    lgh(2) = patch([time(ij), time(ij(end:-1:1))],[yMin(ij), yMax(ij(end:-1:1))],WSettings.colorVector(1,:),'linestyle','none','FaceAlpha',0.5,...
-        'EdgeColor',WSettings.colorVector(1,:),'displayname',strtrim(sprintf('%s %s',legendEntries{1},rangeTxt)));
+    lgh(2) = patch([SimTL.time(ij), SimTL.time(ij(end:-1:1))],[yMin(ij), yMax(ij(end:-1:1))],WSettings.colorVector(1,:),...
+        'linestyle','-','LineWidth',0.5,'FaceAlpha',WSettings.FaceAlpha,...
+        'EdgeColor',WSettings.colorVector(1,:),'displayname',strtrim(strrep(legendEntries{1},'<xxx>',rangeTxt)));
 end
 
-% Patch of referencep poulation
-if ~isnan(yMinRef)
-    
-    % exclude log(0)
-     ij = find(~isinf(yMinRef) & ~isinf(yMaxRef)); 
-
-    lgh(4) = patch([timeRef(ij), timeRef(ij(end:-1:1))],[yMinRef(ij), yMaxRef(ij(end:-1:1))],WSettings.colorVector(2,:),'linestyle','none','FaceAlpha',0.5,...
-        'EdgeColor',WSettings.colorVector(2,:),'displayname',strtrim(sprintf('%s %s',legendEntries{2},rangeTxt)));
-end
 
 % Plot mean
-lgh(1)=plot(time,yMean,'-','color',WSettings.colorVector(1,:),'linewidth',2,...
-    'displayname',strtrim(sprintf('%s %s',legendEntries{1},legendTextMean)));
+lgh(1)=plot(SimTL.time,yMean,'-','color',WSettings.colorVector(1,:),'linewidth',2,...
+    'displayname',strtrim(strrep(legendEntries{1},'<xxx>',legendTextMean)));
 
-% Mean of referencep population
-if ~isnan(yMinRef)
-    lgh(3)=plot(timeRef,yMeanRef,'-','color',WSettings.colorVector(2,:),'linewidth',2,...
-        'displayname',strtrim(sprintf('%s %s',legendEntries{2},legendTextMean)));
+% plotMeanModel
+if ~isempty(SimTL.yMeanModel)
+    lgh(end+1) = plot(SimTL.time,SimTL.yMeanModel,'-.','color',WSettings.colorVector(1,:)*0.7,'linewidth',2,...
+        'displayname',strtrim(legendEntries{4}));
+    
+    % resort legend
+    lgh = [lgh(1:2) lgh(end) lgh(3:end-1)];
 end
+
     
 % plot data
```

Comment:
In case we have a plotted reference simulation with data available, I suggest to plot the respective data in comparison as well. That requires different coloring (and maybe a different type of symbolization).